os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +51,102 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
+static struct mpu_usages_s g_mpu_region_usages;
+
+#ifdef CONFIG_ARMV7M_MPU
+#define MPU_MAX_NREGION		CONFIG_ARMV7M_MPU_NREGIONS
+#elif defined(CONFIG_ARMV8M_MPU)
+#define MPU_MAX_NREGION		CONFIG_ARMV8M_MPU_NREGIONS
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_region_initialize(struct mpu_usages_s *mpu)
+{
+	uint8_t offset = (struct mpu_usages_s *)mpu->nregion_board_specific;
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += MPU_NUM_REGIONS;
+#else
+	offset += MPU_NUM_REGIONS;
+#endif
+	mpu->nregion_common_bin = offset;
+#endif
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += MPU_NUM_REGIONS;
+#else
+	offset += MPU_NUM_REGIONS;
+#endif
+	mpu->nregion_app_bin = offset;
+#endif
+
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	offset += 1;
+	mpu->nregion_stackovf = offset;
+#endif
+	mpu->max_nregion = offset;
+}
+
+/****************************************************************************
+ * Name: mpu_get_region_info
+ *
+ * Description:
+ *   Get the MPU region numbers.
+ *
+ ****************************************************************************/
+
+uint8_t mpu_get_nregion_info(enum mpu_region_usages_e usages)
+{
+	uint8_t nregion = 0;
```

Comment:
Let's add `default` instead of zero initialization. Because it can reduce unnecessary operation in the cases of valid usages.

Reply:
done